cuqipy_fenics/geometry.py

```diff
@@ -398,3 +403,222 @@ def _build_basis(self):
         # Normalize the eigenvectors if required
         if self.normalize:
             self._eig_vec /= np.linalg.norm( self._eig_vec )
+
+
+# Helper user expression to define the step expansion basis
+class _StepExpression(dl.UserExpression):
+    def __init__(self, degree, x_lim, y_lim=None):
+        self.x_lim = x_lim
+        self.y_lim = y_lim
+        super().__init__(degree=degree)
+
+    def eval(self, value, x):
+        if (
+            x[0] > self.x_lim[0] + dl.DOLFIN_EPS
+            and x[0] < self.x_lim[1] - dl.DOLFIN_EPS
+            and (self.y_lim is None or x[1] > self.y_lim[0] + dl.DOLFIN_EPS)
+            and (self.y_lim is None or x[1] < self.y_lim[1] - dl.DOLFIN_EPS)
+        ):
+            value[0] = 1
+        else:
+            value[0] = 0
+
+
+class StepExpansion(_WrappedGeometry):
+    """A geometry class that parameterizes finite element functions on a 1D or
+    2D domain with piecewise constant functions. In the 1D case, the domain is
+    divided into `num_steps_x` constant functions, each function support is of
+    length `L_x`/`num_steps_x`, where `L_x` is the length of the domain in the
+    x direction. In the 2D case, the domain is divided into `num_steps_x`
```

Comment:
Would it possible to document how the basis functions are ordered? I guess x is from left to right and y is from bottom to top?

Reply:
Thank you @chaozg. Good point! added